src/xunit.analyzers/PublicMethodShouldBeMarkedAsTest.cs

```diff
@@ -50,10 +50,13 @@ internal override void AnalyzeCompilation(CompilationStartAnalysisContext compil
                     if (method.MethodKind != MethodKind.Ordinary || method.IsAbstract)
                         continue;
 
-                    var isTestMethod = method.GetAttributes().ContainsAttributeType(xunitContext.Core.FactAttributeType);
+                    var attributes = method.GetAttributes();
+                    var isTestMethod = attributes.ContainsAttributeType(xunitContext.Core.FactAttributeType);
                     hasTestMethods = hasTestMethods || isTestMethod;
 
-                    if (isTestMethod)
+                    if (isTestMethod ||
+                        attributes.Any(attribute => attribute.AttributeClass.GetAttributes()
```

Comment:
Will this still work with attribute class hierarchies where only the base type is annotated with the magical attribute but some derived attribute type is not (but inherits the annotation)? Also, what happens if the attribute is marked to not be inherited via [AttributeUsageAttribute.Inherited](https://msdn.microsoft.com/en-us/library/system.attributeusageattribute.inherited(v=vs.110).aspx)

Reply:
> Will this still work with attribute class hierarchies where only the base type is annotated with the magical attribute but some derived attribute type is not (but inherits the annotation)?

No, but I don't think this worth worrying about. It's trivial to annotate the derived class if required, and I think this somewhat of an edge case anyway.

Reply:
Would you mind adding tests that document this design choice?

Reply:
@marcind I added another commit which adds such a test (I also rebased the whole thing).